emukit/quadrature/kernels/integration_measures.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+
+import numpy as np
+from typing import Tuple, List
+
+
+class IntegrationMeasure:
+    """An integration measure"""
+
+    def __init__(self, name: str):
+        """
+        :param name: Name of the integration measure
+        """
+        self.name = name
+
+    def compute_density(self, x: np.ndarray) -> np.ndarray:
+        """
+        Computes the density at point x
+        :param x: points at which density is computes, shape (num_points, dim)
+        :return: the density at x, shape (num_points, )
+        """
+        raise NotImplementedError
+
+    def get_box(self) -> List[Tuple[float, float]]:
+        """
+        Meaningful box-bounds around the measure. Outside this box, the measure should be virtually zero.
+
+        :return: box in which the measure lies. List of D tuples, where D is the dimensionality and the tuples contain
+        the lower and upper bounds of the box i.e., [(lb_1, ub_1), (lb_2, ub_2), ..., (lb_D, ub_D)]
+        """
+        raise NotImplementedError
+
+
+class UniformMeasure(IntegrationMeasure):
+    """The Uniform measure"""
+
+    def __init__(self, bounds: List[Tuple[float, float]]):
+        """
+        :param bounds: List of D tuples, where D is the dimensionality of the domain and the tuples contain the lower
+        and upper bounds of the box defining the uniform measure i.e., [(lb_1, ub_1), (lb_2, ub_2), ..., (lb_D, ub_D)]
+        """
+        super().__init__('UniformMeasure')
+
+        # checks if lower bounds are smaller than upper bounds.
+        for bounds_d in bounds:
+            lb_d, ub_d = bounds_d
+            if lb_d >= ub_d:
+                raise ValueError("Upper bound of uniform measure must be larger than lower bound. Found a pair "
+                                 "containing (" + str(lb_d) + ", " + str(ub_d) + ").")
+
+        self.bounds = bounds
+        # uniform measure has constant density which is computed here.
+        self.density = self._compute_density()
+
+    def _compute_density(self) -> float:
+        differences = np.array([x[1] - x[0] for x in self.bounds])
+        volume = np.prod(differences)
+
+        if not volume > 0:
+            raise NumericalPrecisionError("Domain volume of uniform measure is not positive. Its value is "
+                                          + str(volume) + ". It might be numerical problems...")
```

Comment:
"It might be numerical problems..." - is it a valid sentence? Does it imply some issues with the input, or that the user may experience issues going forward?

Reply:
I had put this check in case the integration measure is defined on a very small space, e.g., a 10-D integral over the bounds [0, 1e-12]. Then we would multiply 1e-12 10 times which might potentially be zero. This is nothing we can prevent. It is just an ill-conditioned integration problem. But it is easier to debug if there is a check here. 
I deleted the "It might be numerical problems..." part as it does not add additional information.